src/main/scala/ignition/core/cache/ExpiringMultiLevelCache.scala

```diff
@@ -17,9 +17,23 @@ import scala.util.control.NonFatal
 import scala.util.{Failure, Success, Try}
 
 object ExpiringMultiLevelCache {
-  case class TimestampedValue[V](date: DateTime, value: V) {
-    def hasExpired(ttl: FiniteDuration, now: DateTime): Boolean = {
-      date.plus(ttl.toMillis).isBefore(now)
+  case class TimestampedValue[V](date: DateTime,
+                                 value: Option[V] = None,
+                                 hasError: Boolean = false,
+                                 throwable: Option[Throwable] = None) {
+    def hasExpired(ttl: FiniteDuration, now: DateTime, ttlCachedErrors: FiniteDuration = 1.minute): Boolean = {
+      if (!hasError) {
+        date.plus(ttl.toMillis).isBefore(now)
+      } else {
+        date.plus(ttlCachedErrors.toMillis).isBefore(now)
+      }
+    }
+    def getValue: V = {
+      this.value match {
+        case Some(x) => x
+        // We should never try to get a value that was saved as None, probably from an error
+        case None => throw new Exception("Trying to get None value")
```

Comment:
Remember: unhandled exceptions and nulls are your enemies!!

If you **REALLY** want this exception, make sure it's wrapped by a Try or a Future